viadot/utils.py

```diff
@@ -54,13 +63,23 @@ def handle_api_response(
 
         session.mount("http://", adapter)
         session.mount("https://", adapter)
-        response = session.get(
-            url,
-            auth=auth,
-            params=params,
-            headers=headers,
-            timeout=timeout,
-        )
+        if method == "GET":
+            response = session.get(
+                url,
+                auth=auth,
+                params=params,
+                headers=headers,
+                timeout=timeout,
+            )
+        elif method == "POST":
+            response = session.post(
+                url,
+                auth=auth,
+                params=params,
+                headers=headers,
+                timeout=timeout,
+                data=body,
+            )
```

Comment:
You can use `session.request(method=method)`, then you don't need this big elif.